File: yw/dom/Comment.lua
--[[
    Copyright (c) 2024, Oh Inseo (YJK) <dhdlstjtr@gmail.com>
    SPDX-License-Identifier: BSD-3-Clause
    This software may contain third-party material. For more info, see README.
]]
local object        = require "yw.common.object"
local CharacterData = require "yw.dom.CharacterData"


---https://dom.spec.whatwg.org/#comment
---@class DOM_Comment : DOM_CharacterData
local Comment = object.create(CharacterData)

---@param document DOM_Document
---@param data string
---@return DOM_Comment
function Comment:new(document, data)
    local o = CharacterData.new(self, document, data) --[[@as DOM_Comment]]
    o.isComment = true
    return o
end

return Comment

File: dom/comment.go
// This file is part of YW project. Copyright 2025 Oh Inseo (YJK)
// SPDX-License-Identifier: BSD-3-Clause
// See LICENSE for details, and LICENSE_THIRDPARTY for third-party license information.

package dom

import (
	"fmt"
)

// Comment represents a [DOM Comment].
//
// [DOM Comment]: https://dom.spec.whatwg.org/#comment
type Comment interface {
	CharacterData
}
type commentImpl struct {
	CharacterData
}

// NewComment constructs a new [Comment] node.
func NewComment(doc Document, text string) Comment {
	return &commentImpl{newCharacterData(doc, text, CommentCharacterData)}
}
func (cm commentImpl) String() string {
	return fmt.Sprintf("<!-- %s -->", cm.Text())
}
